<?xml version="1.0"?>
<robot xmlns:xacro="http://ros.org/wiki/xacro" name="wam-v-components">
  <xacro:macro name="yaml_components">
    <!-- === lidar === -->
      <xacro:lidar name="lidar" type="16_beam" x="0.7" y="0.0" z="1.6" R="0.0" P="0.1" Y="0.0" />

    <!-- === wamv_camera === -->
      <xacro:wamv_camera name="camera" visualize="False" x="0.75" y="0.0" z="1.5" R="0.0" P="${radians(15)}" Y="0.0" />

    <!-- === wamv_gps === -->
      <xacro:wamv_gps name="gps" x="-0.85" y="0.0" z="1.3" R="0.0" P="0.0" Y="0.0" />

    <!-- === wamv_imu === -->
      <xacro:wamv_imu name="imu" x="-0.8" y="0.0" z="1.3" R="0.0" P="0.0" Y="0.0" />

    <!-- === wamv_ball_shooter === -->
      <xacro:wamv_ball_shooter name="ball_shooter" x="0.55" y="0.1" z="1.3" pitch="${radians(-45)}" yaw="0.0" />

    <!-- === wamv_pinger === -->
      <xacro:wamv_pinger name="pinger" position="1.0 0 -1.0" />

  </xacro:macro>
</robot>